### PR TITLE
Fixed active call notification

### PR DESCRIPTION
**Pull request by afalls-twilio**
## Submission Checklist

 - [x] Updated the `CHANGELOG.md` to reflect any **feature**, **bug fixes**, or **known issues** made in the source code
 - [x] Tested code changes and observed expected behavior in the example app
 - [x] Performed a visual inspection of the `Files changed` tab prior to submitting the pull request for review to ensure proper usage of the style guide

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

## Description

Fixed so that active call notifications are put up when user accepts call from application.

## Breakdown

- Cleaned up flow so that there is only one path for accepting calls, through the service.
- Fixed bug where accepting a call from JS caused the In-Call notification to not show up.

## Validation

- tested

## Additional Notes

None

### CODE REVIEW

**Review by kbagchiGWC — APPROVED** [review]

File: android/src/main/java/com/twiliovoicereactnative/TwilioVoiceReactNativeModule.java [hunk 1]
```diff
@@ -510,40 +510,23 @@ public void onComplete(@NonNull Task<String> task) {
...
-    // Send Event to upstream
-    WritableMap params = Arguments.createMap();
-    WritableMap callInviteInfo = serializeCallInvite(callInviteUuid, activeCallInvite);
-    params.putString(VoiceEventType, VoiceEventCallInviteAccepted);
-    params.putMap(EVENT_KEY_CALL_INVITE_INFO, callInviteInfo);
-    AndroidEventEmitter.getInstance().sendEvent(ScopeVoice, params);
-
-    int notificationId = Storage.uuidNotificaionIdMap.get(callInviteUuid);
+    // Send Event to service
+    final int notificationId = Storage.uuidNotificaionIdMap.get(callInviteUuid);
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/TwilioVoiceReactNativeModule.java:523]:
```suggestion
    final int notificationId = Storage.uuidNotificationIdMap.get(callInviteUuid);
```
I think it's spelled wrong in the `Storage.java` file.

File: android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java [hunk 2]
```diff
@@ -128,6 +130,18 @@ public void onReceive(Context context, Intent intent) {
...
+        if (Storage.callAcceptedPromiseMap.containsKey(uuid)) {
+          Promise promise = Storage.callAcceptedPromiseMap.get(uuid);
+          final Call call = Storage.callMap.get(uuid);
+          final WritableMap callInfo = serializeCall(uuid, call);
+
+          Storage.releaseCallInviteStorage(uuid, callInvite.getCallSid(), Storage.uuidNotificaionIdMap.get(uuid), "accept");
+
+          promise.resolve(callInfo);
+        } else {
+          Storage.releaseCallInviteStorage(uuid, callInvite.getCallSid(), Storage.uuidNotificaionIdMap.get(uuid), "accept");
```
Comment (mhuynh5757) [inline android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java:143]: Is this point in the code the final failstate? Should we reject the promise here?

**Review by mhuynh5757 — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/twiliovoicereactnative/TwilioVoiceReactNativeModule.java (hunk 1 above)
Comment (afalls-twilio) [reply android/src/main/java/com/twiliovoicereactnative/TwilioVoiceReactNativeModule.java:523]: lol, omg! thats embarrassing... thanks!

File: android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java (hunk 2 above)
Comment (afalls-twilio) [reply android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java:143]: Not all 'accept' call flows are paired with a promise object, for example, when a notification action/button is selected.

Comment (mhuynh5757) [reply android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java]: Ah I see. None of those functions can throw right? I'm concerned if there's an edge case such that the Promise is never resolved or rejected, and the JS layer hangs.

Comment (afalls-twilio) [reply android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java]: its low probability, but good call out.. its fixed now.

**Review by mhuynh5757 — APPROVED** [review]

**Pull request merged**